### PR TITLE
Adding newWithResizeX

### PR DESCRIPTION
**Pull request by interesaaat**
This is needed to implement Resize in SNT #13

### CODE REVIEW

File: TorchSharp/TypeGeneration.tt [hunk 1]
```diff
@@ -198,24 +198,23 @@ foreach (var type in new [] {
         {
             handle = TH<#=tname#>Tensor_new ();
         }
-        
-        internal <#=tname#>Tensor (HType handle)
+
+                internal <#=tname#>Tensor (HType handle)
```
Comment (motus) [inline TorchSharp/TypeGeneration.tt]: why do we indent like that?

File: TorchSharp/TypeGeneration.tt [hunk 2]
```diff
@@ -198,24 +198,23 @@ foreach (var type in new [] {
...
         {
             this.handle = handle;
         }
-        
-        [DllImport ("caffe2")]
+		[DllImport ("caffe2")]
```
Comment (motus) [inline TorchSharp/TypeGeneration.tt]: fix the indentation

File: TorchSharp/TypeGeneration.tt [hunk 3]
```diff
@@ -198,24 +198,23 @@ foreach (var type in new [] {
...
         extern static HType TH<#=tname#>Tensor_newWithSize1d (long size0);
+
         /// <summary>
         ///    Creates a 1D tensor of the specified size.
         /// </summary>    
         /// <param name="size0">Size for the first dimension.</param>
-
-        public <#=tname#>Tensor (long size0)
+         public <#=tname#>Tensor (long size0)
```
Comment (motus) [inline TorchSharp/TypeGeneration.tt]: indentation

File: TorchSharp/TypeGeneration.tt [hunk 4]
```diff
@@ -472,6 +471,74 @@ foreach (var type in new [] {
...
+        ///   Access to element at the specified position in the tensor
+        /// </summary>        
+        /// <param name="offset">Offset within the input storage the storage of the new tensor will start from.</param> 
+        /// <param name="size0">Size of the first dimension.</param>     
+        /// <param name="stride0">Stride of the first dimension.</param>
+        /// <param name="size1">Size of the second dimension.</param>     
+        /// <param name="stride1">Stride of the second dimension.</param>
+        /// <param name="size2">Size of the third dimension.</param>     
+        /// <param name="stride2">Stride of the third dimension.</param>
+        /// <param name="size3">Size of the third dimension.</param>
```
Comment (motus) [inline TorchSharp/TypeGeneration.tt]: Must be **fourth** dimension, right?

File: TorchSharp/TypeGeneration.tt [hunk 5]
```diff
@@ -472,6 +471,74 @@ foreach (var type in new [] {
...
+        /// </summary>        
+        /// <param name="offset">Offset within the input storage the storage of the new tensor will start from.</param> 
+        /// <param name="size0">Size of the first dimension.</param>     
+        /// <param name="stride0">Stride of the first dimension.</param>
+        /// <param name="size1">Size of the second dimension.</param>     
+        /// <param name="stride1">Stride of the second dimension.</param>
+        /// <param name="size2">Size of the third dimension.</param>     
+        /// <param name="stride2">Stride of the third dimension.</param>
+        /// <param name="size3">Size of the third dimension.</param>     
+        /// <param name="stride3">Stride of the third dimension.</param>
```
Comment (motus) [inline TorchSharp/TypeGeneration.tt]: same as above

File: TorchSharp/TypeGeneration.tt (hunk 4 above)
Comment (interesaaat) [reply TorchSharp/TypeGeneration.tt]: Good catch!

File: TorchSharp/TypeGeneration.tt (hunk 1 above)
Comment (interesaaat) [reply TorchSharp/TypeGeneration.tt]: No idea. VS added tabs instead of spaces although I changed in my default settings :( Anyway fixed

**Comment by migueldeicaza** [comment]
This looks great!

**Pull request merged**